File: library/src/main/java/com/ebr163/inspector/rule/TextNotEmptyRule.java
package com.ebr163.inspector.rule;

/**
 * Created by Bakht
 * on 09.01.2018.
 */

public class TextNotEmptyRule extends AbstractRule<CharSequence> {

    public TextNotEmptyRule(String errorMessage) {
        super(errorMessage);
    }

    @Override
    public boolean verify(CharSequence value) {
        return value != null && value.length() != 0;
    }
}

File: library/src/main/java/com/ebr163/inspector/rule/TextNotEmptyRule.kt
package com.ebr163.inspector.rule

/**
 * Created by Bakht
 * on 09.01.2018.
 */

class TextNotEmptyRule(errorMessage: String) : AbstractRule<String>(errorMessage) {

    override fun verify(value: String?): Boolean = !value.isNullOrEmpty()
}